### ISSUE TITLE
Not all custom fields are visible in the transaction dialog

### ISSUE THREAD

**Issue by cmbarbu** [moneymanagerex/moneymanagerex#4314]
MMEX version:
 - [x] 1.5.14
 - [ ] 1.6.x

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

I have defined more than 5 custom fields. The ones that do not have a column assigned to them (`Panel's columns: UDFC...`) are not visible in the transaction dialog. This was not the case in previous versions (< 1.5.14).

![mmex-bug-custom-fields-not-visible-2](https://user-images.githubusercontent.com/1184162/166257411-bd81a4b8-104b-4859-a952-18ca7a36d354.PNG)
![mmex-bug-custom-fields-not-visible-1](https://user-images.githubusercontent.com/1184162/166257415-73aeb6c8-c6d1-44d2-9078-5e0e31f7028e.PNG)
![mmex-bug-custom-fields-not-visible-3](https://user-images.githubusercontent.com/1184162/166257606-5a475f26-58a6-4740-b75a-d8bc1b47cc3d.PNG)

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

**Comment by cmbarbu** [comment]
Note that since there are only 5 custom columns available, not every custom field can have one defined.

**Comment by whalley** [comment]
Yes, seems like a bug. If we are allowing more than 5 custom fields to be defined, but with only 5 visible in the account views then we should allow all fields to be edited against the transaction.

**Comment by whalley** [comment]
@vomikan - Yes, with the Transaction Report filter it should be possible to filter on any custom field irrespective of whether it is displayed in the report. 

This change allows this, but the saving/restoring based on UDF01 - UDF05 causes issues. I suggest we save/fetch the custom fields in JSON indexed by FIELDID rather than PROPERTIES.

Guess we should raise another issue (unless we already have one that I've missed) to capture that we should enhance the report output to allow the UDF01 - UDF05 entries to be presented.

**Comment by vomikan** [comment]
> This change allows this, but the existing saving/restoring based on UDF01 - UDF05 causes issues. I suggest we save/fetch the custom fields in JSON indexed by FIELDID rather than PROPERTIES.

Absolutely! I couldn't remember it yesterday.
